src/api.ts

```diff
@@ -792,17 +800,40 @@ function createApi({
         style: styleToSave,
       })
     },
-
     async listStyles() {
+      // `bytesStored` calculates the total bytes stored by tiles that the style references
+      // Eventually we want to get storage taken up by other resources like sprites and glyphs
       return db
         .prepare(
-          "SELECT Style.id, json_extract(stylejson, '$.name') as name FROM Style"
+          `
+          SELECT Style.id, 
+            json_extract(Style.stylejson, '$.name') as name, 
+            (
+              SELECT SUM(LENGTH(TileData.data))
+              FROM TileData
+              JOIN Tile ON Tile.tileHash = TileData.tileHash
+              JOIN Tileset ON Tileset.id = Tile.tilesetId
+              JOIN (
+                SELECT S2.id AS styleId, StyleReferencedTileset.value AS tilesetId
+                FROM Style S2, json_each(S2.sourceIdToTilesetId, '$') AS StyleReferencedTileset
+              ) AS SourceTileset ON SourceTileset.tilesetId = Tileset.id
+              JOIN Style S1 ON S1.id = SourceTileset.styleId
+              WHERE SourceTileset.styleId = Style.id
+            ) AS bytesStored
+          FROM Style;
+          `
         )
         .all()
-        .map((row: { id: string; name?: string }) => ({
-          ...row,
-          url: getStyleUrl(row.id),
-        }))
+        .map(
+          (row: {
+            id: string
+            bytesStored: number | null
```

Comment:
What would cause `bytesStored` to be `null`? Does it make sense to return `0` instead of `null` to make the API simpler to deal with?

Reply:
an example would be styles that are created without importing a tileset. probably no real-world flow that we support that leads to this at the moment. 

for the sake of simplicity, can default to 0 if it is null

Reply:
Yes, let's default to `0` since in the example you give, `null` would be bytesStored = 0 anyway.